modules/Bio/EnsEMBL/Compara/RunnableDB/PrepareMaster/UpdateGenomesFromMetadataFactory.pm

```diff
@@ -255,4 +300,122 @@ sub fetch_genome_report {
     return ([@new_genomes, @updated_assemblies, @updated_annotations], \%renamed_genomes, \@updated_annotations);
 }
 
+# Internal method to find the current rapid release number and dba
+sub _get_curr_rr_release {
+    my ($name, $host) = @_;
+
+    my $port = Bio::EnsEMBL::Compara::Utils::Registry::get_port( $host );
+    my $dba  = Bio::EnsEMBL::MetaData::DBSQL::MetaDataDBAdaptor->new(
+        '-species' => 'multi',
+        '-group'   => 'metadata',
+        '-host'    => $host,
+        '-port'    => $port,
+        '-user'    => "ensro",
+        '-pass'    => "",
+        '-dbname'  => $name, );
+    my $helper = $dba->dbc->sql_helper;
+
+    my $sql = 'SELECT MAX(ensembl_genomes_version) FROM data_release WHERE is_current = 1';
+    # should only return one result, as there is only one current at a time
+    my $rr  = $helper->execute_single_result(-SQL => $sql);
+
+    return ($rr, $dba);
+}
+
+sub _get_rr_reference_genomes {
+    my ( $self, $metadata_dba, $meta_host, $rr, $report_exe, $dump_path ) = @_;
+
+    # this is the rapid release number
+    my $metadata_name = $self->param('rr_meta_name');
+
+    my $meta_options = "\$($meta_host details script) --dbname $metadata_name --release $rr --eg_first 1 --dump_path $dump_path";
+
+    my $cmd = "perl $report_exe $meta_options";
+    $self->run_command($cmd);
+
+    # script produces a per-division tsv text file - not support for json yet
+    my @new_genomes_files         = glob($dump_path . "/*-new_genomes.txt");
+    my @updated_assemblies_files  = glob($dump_path . "/*-updated_assemblies.txt");
+    my @updated_annotations_files = glob($dump_path . "/*-updated_annotations.txt");
+    my @renamed_genomes_files     = glob($dump_path . "/*-renamed_genomes.txt");
+
+    my ( @new_genomes, @updated_assemblies, @updated_annotations, %renamed_genomes );
+    foreach my $file ( @new_genomes_files ) {
+        push @new_genomes, _parse_name_to_array($file);
+    }
+    foreach my $file ( @updated_assemblies_files ) {
+        push @updated_assemblies, _parse_name_to_array($file);
+    }
+    foreach my $file ( @updated_annotations_files ) {
+        push @updated_annotations, _parse_name_to_array($file);
+    }
+    %renamed_genomes = %{_parse_rename_rr_files(\@renamed_genomes_files)};
+
+    return \@new_genomes, \@updated_assemblies, \@updated_annotations, \%renamed_genomes;
+}
+
+# Internal method to parse the report file generated by report_genomes_exe for rr - as not json compatible
+sub _parse_rr_genome_files {
```

Comment:
Not really a fan of having this method **before** the function that calls it, but can be left as it is :nerd_face:

Reply:
I thought I had already moved it actually, but it hopped back. All by itself.